### ISSUE TITLE
ClassVar[Final[T]] should be allowed

### ISSUE THREAD

**Issue by ben-alexander** [microsoft/pyright#10500]
Pyright doesn't allow the construction `Final[ClassVar[T]]`, [`typing.ClassVar`](https://docs.python.org/3/library/typing.html#typing.ClassVar) states:

> Changed in version 3.13: [ClassVar](https://docs.python.org/3/library/typing.html#typing.ClassVar) can now be nested in [Final](https://docs.python.org/3/library/typing.html#typing.Final) and vice versa.

```python
from typing import ClassVar, Final

class A:
    var: ClassVar[str] = 'a'

class B(A):
    var: Final[ClassVar[str]] = 'b'

# "ClassVar" is not allowed in this context Pylance(reportInvalidTypeForm)
```

There was a previous report https://github.com/microsoft/pyright/issues/7959 specific to dataclasses from before the release of 3.1.3. I'm not sure where this restriction to dataclasses comes from as the typing spec doesn't mention this in 3.1.12.

Running Pylance in VS Code. Version: 2025.5.1

**Comment by erictraut** [comment]
Thanks, I agree this is a bug. There's actually two bugs demonstrated in your code. The first is that pyright generates a "ClassVar not allowed in this context" diagnostic. The second is that pyright doesn't generate a diagnostic for attempting to override a non-final attribute with a final attribute. That combination violates the Liskov Substitution Principle.
